src/Diagnostics.DataProviders/AscClient.cs

```diff
@@ -231,9 +236,9 @@ private static T CastTo<T>(object obj)
 
         private async Task<T> GetAscResponse<T>(HttpRequestMessage requestMessage, bool isBlobRequest, CancellationToken cancellationToken = default(CancellationToken))
         {
-            if (!string.IsNullOrWhiteSpace(SubscriptionLocationPlacementId) && SubscriptionLocationPlacementId.Equals("geos_2020-01-01", StringComparison.CurrentCultureIgnoreCase))
+            if (!string.IsNullOrWhiteSpace(SubscriptionLocationPlacementId) && SubscriptionLocationPlacementId.Equals(DiagAscHeaderValue, StringComparison.CurrentCultureIgnoreCase))
             {
-                throw new HttpRequestException(MethodNotAllowed);
+                throw new InvalidOperationException();
```

Comment:
Can we add a message saying something like "this subscription is not allowed for ASC calls". That way we can just show that message in the markdown message in the detector UI